test/Microsoft.AspNetCore.DataProtection.Redis.Test/DataProtectionRedisTests.cs

```diff
@@ -55,5 +67,33 @@ public void StoreElement_PushesValueToList()
 
             database.Verify();
         }
+
+        [ConditionalFact]
+        [TestRedisServerIsAvailable]
+        public async Task XmlRoundTripsToActualRedisServer()
+        {
+            var connStr = TestRedisServer.GetConnectionString();
+            _output.WriteLine("Attempting to connect to " + connStr);
+            using (var redis = await ConnectionMultiplexer.ConnectAsync(connStr).TimeoutAfter(TimeSpan.FromMinutes(1)))
+            {
+                var guid = Guid.NewGuid().ToString();
+                RedisKey key = "Test:DP:Key" + guid;
+
+                try
+                {
+                    var repo = new RedisXmlRepository(() => redis.GetDatabase(), key);
```

Comment:
I would recreate the repo in between store and GetAllElements

Reply:
Good suggestion.